PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -807,22 +823,26 @@ EFIAPI
 LoadBootImages (
   IN  OS_BOOT_OPTION  *OsBootOption,
   IN  EFI_HANDLE       HwPartHandle,
-  IN  EFI_HANDLE       FsHandle,
   OUT EFI_HANDLE      *LoadedImageHandle
   )
 {
   LOADED_IMAGES_INFO        *LoadedImagesInfo;
   LOADED_IMAGE              *LoadedImage;
   BOOT_IMAGE                *BootImage;
   UINT8                      BootFlags;
+  BOOLEAN                    BootFromFs;
   EFI_STATUS                 Status;
   UINT8                      Index;
   CONTAINER_IMAGE           *ContainerImage;
+  EFI_HANDLE                 FsHandle;
 
   ASSERT (OsBootOption != NULL);
 
   BootImage = OsBootOption->Image;
   BootFlags = OsBootOption->BootFlags;
+  BootFromFs = (OsBootOption->DevType != OsBootDeviceSpi) &&
+               (OsBootOption->DevType != OsBootDeviceMemory) &&
+               (OsBootOption->FsType < EnumFileSystemMax);
```

Comment:
It looks the default FsType is 0 (EnumFileSystemTypeFat), so we could not use FsType to get BootFromFs.

Reply:
Here `OsBootOption` is `OS_BOOT_OPTION`, its `FsType` comes directly from boot option. `FILE_IMAGE_LOCATION->FsType` is not used here.

Reply:
In one boot option, SBL already supports to load raw image from different partition. With this change, SBL could support to load image by file name from different partition. So it is possible in in boot option some images are loaded by file name, and other image are loaded by LBA address. SBL might read each image differently, so the FsType and SwPart from Os_BOOT_OPTION should not be used, or it is used only for the normal boot image.

Reply:
Got it.